src/main/java/com/zenith/network/server/handler/spectator/incoming/movement/PlayerPositionRotationSpectatorHandler.java

```diff
@@ -29,31 +31,47 @@ public static void updateSpectatorPosition(final ServerConnection selfSession) {
         if (selfSession.isPlayerCam()) {
             return;
         }
+        double playerEyeHeight = 1.62;
+        EntityStats entityStats = new EntityStats();
```

Comment:
Integrate any additional metadata about spectator entity types in the SpectatorEntity class hierarchy under here: https://github.com/rfresh2/ZenithProxy/tree/1.20/src/main/java/com/zenith/feature/spectator

The current spectator entity type can be accessed here on the ServerConnection: https://github.com/rfresh2/ZenithProxy/blob/c04b583187cbb0dc5d745fd4bb5ec562f804cae6/src/main/java/com/zenith/network/server/ServerConnection.java#L78

so you shouldn't need any `EntityData` class initializations or hardcoded map lookups in the `updateSpectatorPosition` function.

I'm imagining we add a set of methods on SpectatorEntity to get whatever data is needed about the current type.

Reply:
change was made